packages/scaffold-config/src/frameworks.ts

```diff
@@ -14,10 +14,42 @@ export type WizardBundler = typeof dependencies.WIZARD_BUNDLERS[number]
 
 export type CodeGenFramework = Cypress.ResolvedComponentFrameworkDefinition['codeGenFramework']
 
+const yarnPnpRegistrationPath = new Map<string, { usesYarnPnP: boolean }>()
+
+async function readPackageJson (packageFilePath: string, projectPath: string): Promise<PkgJson> {
+  if (yarnPnpRegistrationPath.get(projectPath)?.usesYarnPnP) {
+    // using Yarn PnP. You cannot `fs.readJson`, since the module is zipped.
+    // use require.resolve - The PnP runtime (.pnp.cjs) automatically patches Node's
+    // fs module to add support for accessing files inside Zip archives.
+    // @see https://yarnpkg.com/features/pnp#packages-are-stored-inside-zip-archives-how-can-i-access-their-files
+    return require(require.resolve(packageFilePath))
+  }
+
+  return await fs.readJson(packageFilePath)
```

Comment:
Maybe I'm missing an edge case, but couldn't we just use the `require(require.resolve(packageFilePath))` logic in both cases? PnP handles if the module is zipped, and if we aren't in PnP it will be a flat file that should be normally require-able

Reply:
I think there was an edge case where this doesn't work, let me double check.